lib/pbench/agent/tool_data_sink.py

```diff
@@ -632,69 +798,172 @@ def _change_tm_tracking(self, curr, new):
             ), f"_change_tm_tracking unexpected tm posted value, {tm!r}"
             tm["posted"] = new
 
-    def state_change(self, data):
-        """state_change - give a data dictionary, change the state for this
-        data sink instance.
+    def _forward_tms(self, data):
+        """_forward_tms - for the action data payload to the known Tool Meisters.
+
+        Log messages will be posted if any errors were encountered.
+
+        Return 0 on success, non-zero to indicate an error.
+        """
+        self.logger.debug("publish %s", self._to_tms_channel)
+        try:
+            num_present = self.redis_server.publish(
+                self._to_tms_channel, json.dumps(data, sort_keys=True)
+            )
+        except Exception:
+            self.logger.exception("Failed to publish action message to TMs")
+            ret_val = 1
+        else:
+            self.logger.debug("published %s", self._to_tms_channel)
+            if num_present != self._num_tms:
+                self.logger.error(
+                    "TM action message received by %d subscribers", num_present
+                )
+                ret_val = 1
+            else:
+                self.logger.debug("posted TM action message, %r", data)
+                ret_val = 0
+        return ret_val
+
+    def _wait_for_tms(self):
+        """_wait_for_tms - Wait for all the Tool Meisters to report back the
+        status of their action.
+
+        Returns 0 on success, non-zero to indicate an error.
+        """
+        # Wait for an operational status message from the Tool Data Sink
+        # reporting the combined response of the Tool Meisters.
```

Comment:
Replace comment with, "Wait for all Tool Meisters to report back their operational status."